core/src/test/scala/org/apache/spark/sql/delta/cdc/MergeCDCSuite.scala

```diff
@@ -187,6 +193,118 @@ class MergeCDCSuite extends MergeIntoSQLSuite with DeltaColumnMappingTestUtils {
     confs = (DeltaSQLConf.DELTA_SCHEMA_AUTO_MIGRATE.key, "true") :: Nil
   )
 
+
+  testMergeCdcUnlimitedClauses("unconditional delete only with duplicate matches")(
+    target = Seq(0, 1).toDF("value"),
+    source = Seq(1, 1).toDF("value"),
+    mergeCondition = "t.value = s.value",
+    clauses = MergeClause(isMatched = true, null, "DELETE") :: Nil,
+    expectedTableData = Seq(0).toDF(),
+    expectedCdcData = ((1, "delete", 1) :: Nil).toDF()
+  )
+
+  testMergeCdcUnlimitedClauses(
+    "unconditional delete only with duplicate matches without duplicates in the source")(
+    target = Seq(0).toDF("value"),
+    source = ((0, 0) :: (0, 1) :: Nil).toDF("col1", "col2"),
+    mergeCondition = "t.value = s.col1",
+    clauses = MergeClause(isMatched = true, null, "DELETE") :: Nil,
+    expectedTableData =
+      Nil.asInstanceOf[List[Integer]]
+        .toDF("value"),
+    expectedCdcData = ((0, "delete", 1) :: Nil).toDF()
+  )
+
+  testMergeCdcUnlimitedClauses(
+    "unconditional delete only with duplicate matches with duplicates in the target")(
+    target = Seq(0, 1, 1).toDF("value"),
+    source = Seq(1, 1).toDF("value"),
+    mergeCondition = "t.value = s.value",
+    clauses = MergeClause(isMatched = true, null, "DELETE") :: Nil,
+    expectedTableData = Seq(0).toDF(),
+    expectedCdcData = ((1, "delete", 1) :: (1, "delete", 1) :: Nil).toDF()
+  )
+
+  testMergeCdcUnlimitedClauses("unconditional delete only with target-only condition")(
```

Comment:
confusion.. which condition .. clause or main merge one? 
```suggestion
  testMergeCdcUnlimitedClauses("unconditional delete only with target-only merge condition")(
```

Reply:
update the others below

Reply:
oh yeah agree thanks